### PR TITLE
proof-of-concept: account balance tracking

### PR DESCRIPTION
**Pull request by bryanchriswhite**

### CODE REVIEW

File: src/mappings/mappingHandlers.ts [hunk 1]
```diff
@@ -199,3 +196,101 @@ export async function handleLegacyBridgeSwap(msg: CosmosMessage<LegacyBridgeSwap
...
+    spender: string
+    amount: string
+  } = event.event.attributes.reduce(
+    (acc, next) => Object.assign(acc, next),
+    {spender: "", amount: ""}
+  )
+
+  // NB: amountString: "5000000atestfet"
+  const {spender, amount: amountString} = attrs;
+  const [_, amount, denom] = amountString.match(/(\d+)(\w+)/);
```
Comment (bryanchriswhite) [inline src/mappings/mappingHandlers.ts:217]: Don't do this. Use the `parseCoins` function recently added to utils.ts

File: src/mappings/mappingHandlers.ts [hunk 2]
```diff
@@ -199,3 +196,101 @@ export async function handleLegacyBridgeSwap(msg: CosmosMessage<LegacyBridgeSwap
...
+      genesisBalance: BigInt(-1),
+      denom,
+    });
+
+    await nativeBalanceEntity.save();
+  } else {
+    nativeBalanceEntity.balanceOffset -= BigInt(amount);
+  }
+  await accountEntity.save();
+  await nativeBalanceEntity.save()
```
Comment (bryanchriswhite) [inline src/mappings/mappingHandlers.ts:246]:
```suggestion
    nativeBalanceEntity.balanceOffset -= BigInt(amount);
    await nativeBalanceEntity.save()
  }
  await accountEntity.save();
```

File: src/mappings/mappingHandlers.ts [hunk 3]
```diff
@@ -199,3 +196,101 @@ export async function handleLegacyBridgeSwap(msg: CosmosMessage<LegacyBridgeSwap
...
+      genesisBalance: BigInt(-1),
+      denom,
+    });
+
+    await nativeBalanceEntity.save();
+  } else {
+    nativeBalanceEntity.balanceOffset += BigInt(amount);
+  }
+  await accountEntity.save();
+  await nativeBalanceEntity.save()
```
Comment (bryanchriswhite) [inline src/mappings/mappingHandlers.ts:295]:
```suggestion
    nativeBalanceEntity.balanceOffset += BigInt(amount);
    await nativeBalanceEntity.save()
  }
  await accountEntity.save()
```

File: src/mappings/mappingHandlers.ts [hunk 4]
```diff
@@ -284,3 +287,101 @@ export async function handleDelegatorWithdrawRewardEvent(event: CosmosEvent): Pr
...
+
+  // NB: convert [{key: "value"}, {key2: "value2"} ...]
+  //           to {key: "value", key2: "value2" ...}
+  const attrs: {
+    spender: string
+    amount: string
+  } = event.event.attributes.reduce(
+    (acc, next) => Object.assign(acc, next),
+    {spender: "", amount: ""}
+  )
```
Comment (bryanchriswhite) [inline src/mappings/mappingHandlers.ts:304]: This should probably be a helper funciton.

**Comment by daeMOn63** [comment]
completed in #88

**Pull request closed without merging**